mesa: 22.1.7 -> 22.2.1

PedroHLC opened the pull request:
###### Description of changes

- Bumped version;
- Rebased patches;
- Remove upstreamed patches;
- Includes https://github.com/NixOS/nixpkgs/pull/192243#issuecomment-1261242034
- Fixes https://github.com/NixOS/nixpkgs/issues/193655
- Rebases https://github.com/NixOS/nixpkgs/pull/195682 to staging

###### Things done

Built `mesa`, `pkgsMusl.mesa`, `pkgsi686Linux.mesa`.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

PedroHLC committed:
mesa: 22.1.7 -> 22.2.1

PedroHLC commented on pkgs/development/libraries/mesa/default.nix:
```diff
@@ -5,9 +5,10 @@
+, vulkan-loader, glslang
 , galliumDrivers ? ["auto"]
 , vulkanDrivers ? ["auto"]
+, videoCodecs ? [ "h264dec" "h264enc" "h265dec" "h265enc" "vc1dec" ]
```
@jtojnar what do you think, list or `withPatentedCodecs ? true`?

jtojnar replied:
I would guess people would either want to disable or enable all encumbered codecs but I do not have a preference.

I worry about this list getting desynchronized. Especially when I did not see any mention of this change in the release notes, and was only aware of it because someone on GNOME IRC posted a link about Fedora dealing with this.

But I am not sure how to solve that other than asking upstream to add `do-not-care-about-stupid-american-laws' option to the [option](https://gitlab.freedesktop.org/mesa/mesa/-/blob/0c87909bcfeb2283d4326ef8a411598872802486/meson_options.txt#L541).

Or adding a `passthru.tests` script checking that the value we are passing includes all the listed options similarly to what we do e.g. in [fwupd](https://github.com/NixOS/nixpkgs/blob/e6896c17ca6d7897683eff69290b4dc1eb58a8b5/pkgs/os-specific/linux/firmware/fwupd/default.nix#L363-L383).

PedroHLC replied:
I think I can write the test and add it in a future PR, but this I'll have to ask right now:
- Let's move it to a boolean, then?

vcunat replied:
Sounds OK to me.

vcunat reviewed (approved):
Sounds good enough for `staging`.

I don't mind how exactly the codec parameter is exposed, so if someone feels strongly about that...  but I expect it would be rare to override it.

vcunat commented:
For historical reference, we used to have `enableTextureFloats ? false` (also patent stuff, expired now).

PedroHLC committed:
mesa: re-enable video-codecs

The release disables some codecs https://gitlab.freedesktop.org/mesa/mesa/-/commit/7d969fe9e91e39e03041cdfac69bf33337bc2c96.
Since we are not Fedora, we should probably enable them https://lists.fedoraproject.org/archives/list/legal@lists.fedoraproject.org/thread/M4LTGLHY5JX42IHC45WNWB5FH2JIFMAS/.

vcunat commented:
OK, we can amend the parametrization later even without causing any rebuilds.  I'm merging already, so it's included in the upcoming rebuilds.
